tools/cli/domainCommands.go

```diff
@@ -284,10 +284,24 @@ func (d *domainCLIImpl) UpdateDomain(c *cli.Context) {
 func (d *domainCLIImpl) DeprecateDomain(c *cli.Context) {
 	domainName := getRequiredGlobalOption(c, FlagDomain)
 	securityToken := c.String(FlagSecurityToken)
+	force := c.Bool(FlagForce)
 
 	ctx, cancel := newContext(c)
 	defer cancel()
 
+	if !force {
+	// check if there is any workflow in this domain, if exists, do not deprecate
+		wfs, _ := listClosedWorkflow(getWorkflowClient(c), 1, 0, time.Now().UnixNano(), "", "", workflowStatusNotSet, nil, c)
```

Comment:
do we need to check the error and block deprecation when failed to verify workflows?

Reply:
Yes, it's done within the helper functions.

Reply:
Oh I see. I thought the second return value is error. My bad.